src/pages/audio-rewards-page/store/sagas.ts

```diff
@@ -284,6 +297,25 @@ export function* watchFetchUserChallenges() {
   })
 }
 
+function* watchUpdateOptimisticListenStreak() {
+  yield takeEvery(updateOptimisticListenStreak.type, function* () {
```

Comment:
why not just have this watch `socialActions.RECORD_LISTEN`?

Reply:
We had a talk about re-using actions vs making new specified actions with new intentions a while ago and decided to make new actions - I'd actually prefer to use record_listen, but was trying to go along the make a new action model

Reply:
I think if we're going full separate actions, we'd need to fire `socialActions.RECORD_LISTEN` and `updateOptimisticListenStreak` from the same place, rather than firing one from the other?